src/Microsoft.Sbom.Targets/Microsoft.Sbom.Targets.targets

```diff
@@ -0,0 +1,82 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <GenerateSbom_TFM Condition=" '$(MSBuildRuntimeType)' == 'Full' ">net472</GenerateSbom_TFM>
+    <GenerateSbom_TFM Condition=" '$(MSBuildRuntimeType)' == 'Core' ">net8.0</GenerateSbom_TFM>
+
+    <SbomToolBinaryOutputPath Condition="'$(SbomToolBinaryOutputPath)' == ''">$([System.IO.Path]::Combine($(MSBuildThisFileDirectory),..,tasks,$(GenerateSbom_TFM),sbom-tool))</SbomToolBinaryOutputPath>
+    <GenerateSbomTaskAssemblyFilePath Condition="'$(GenerateSbomTaskAssemblyFilePath)' == ''">$([System.IO.Path]::Combine($(MSBuildThisFileDirectory),..,tasks,$(GenerateSbom_TFM),Microsoft.Sbom.Targets.dll))</GenerateSbomTaskAssemblyFilePath>
+
+    <!--Set the SBOM CLI Tool path. This variable is only used in SbomCLIToolTask.cs-->
+    <SbomToolPath Condition=" '$(MSBuildRuntimeType)' == 'Full'">$(SbomToolBinaryOutputPath)</SbomToolPath>
+    <ManifestFolderName>_manifest</ManifestFolderName>
+    <SbomSpecification>spdx_2.2</SbomSpecification>
+  </PropertyGroup>
+  
+  <!--Based on the MSBuild runtime, GenerateSbom will either pull the GenerateSbomTask or SbomCLIToolTask logic-->
+  <UsingTask TaskName="Microsoft.Sbom.Targets.GenerateSbom" AssemblyFile="$(GenerateSbomTaskAssemblyFilePath)" />
+
+  <PropertyGroup>
+    <GenerateSBOM Condition=" '$(GenerateSBOM)' == '' ">false</GenerateSBOM>
+    <SbomGenerationBuildComponentPath Condition=" '$(SbomGenerationBuildComponentPath)' == '' ">$(MSBuildProjectDirectory)</SbomGenerationBuildComponentPath>
```

Comment:
The project directory will be source controlled.  Is the SBOM something that you believe should be checked into source control? Usually, content that is automatically is generated is not commit into source control. In which case, you should write to the intermediate directory, or whatever the `obj` directory's property name is.

Reply:
No worries! The SbomGenerationBuildComponentPath represents a path to a project's .csproj file, and the sbom API uses this to determine dependencies to include in the SBOM. We did do a mini update to our SBOM generation logic as follows:

Dotnet pack -> unzip the .nupkg to a temp directory -> generate a single SBOM for the contents of the temp dir -> re-zip the folder to a .nupkg

Reply:
So, is `SbomGenerationBuildComponentPath` something that customers are expected to set?

The PR description lists it in a table of properties, the others of which sound like they are things that customers will set to customize behaviour. I think the readme in the project directory has the same time.

I don't know if it's documented anywhere, but the general MSBuild convention (given the lack of scopes and therefore private variables) is to use property names starting with an underscore `_` for things customers shouldn't set/change, and everything else are potential customizations for customers.

My feedback is that removing things in the future is a breaking change, which can impact customers who use your product in unexpected ways. Whereas adding things is not a breaking change if you find your initial release needs more features, or customers with edge cases that weren't originally considered need. So, consider the tradeoff, because we (NuGet) have been bitten by overly generic features having problems, then when we tried to fix the problem that customer A reported, customer B reported that the change broke them, making it much harder to implement features that work for both customers A & B.

Reply:
Thank you for this insight! All the properties listed in the README/PR description are available for the customer to set/customize, but we do have default values set for them in case they aren't overriden. We will definitely take note of preventing breaking changes/changes that require removing things